en/ros/ros1_via_ros2.md

```diff
@@ -0,0 +1,131 @@
+# ROS (1) via ROS 2 Bridge (User Guide)
+
+This topic explains how use ROS (1) with PX4, by bridging via [ROS 2](..ros/ros2.md).
+
+It provides an overview of the ROS-ROS2-PX4 architecture, along with instructions on how to install all the needed software and build ROS (1) applications.
+It also covers how to run/setup ROS 2 and ROS 1 workspaces at the same time.
+
+:::note
+Generally you might use this setup rather than bridging [ROS (1) with MAVROS](../ros/ros1.md) if you want deeper access to PX4 than granted by MAVLink, or if you want to use both ROS2 and ROS (1) applications.
+:::
+
+:::note
+This setup, and these instructions *depend* on [ROS 2](../ros/ros2.md).
+We highly recommend you read up on ROS 2 first.
+:::
+
+:::warning Note
+The PX4 development team recommend that all users [upgrade to ROS 2](../ros/ros2.md).
+:::
+
+## Overview
+
+The application pipeline for ROS (1) bridged over ROS 2 is shown below.
+
+![Architecture with ROS](../../assets/middleware/micrortps/architecture_ros.png)
+
+Essentially it is the same as for ROS 2, except that there is an additional [`ros1_bridge`](https://github.com/ros2/ros1_bridge) package (by Open Robotics) that translates messages between the two versions.
+This is needed because the original version of ROS does not support RTPS.
+
+The other main difference is that the `px4_ros_com` and `px4_msgs` packages a separate `ros1` branch that generates the ROS message headers and source files for using **with** the `ros1_bridge`.
+This branch also includes example listener and advertiser nodes.
+
+
+## Installation & Setup
+
+Follow the instructions in [ROS 2 User Guide > Installation & Setup](../ros/ros2.md#installation-setup) to install ROS 2.
+
+### Build ROS (1) Workspace
+
+Since the ROS requires a different environments from ROS to you will need to create a separate workspace.
+This will include the `ros` branch of `px4_ros_com` and `px4_msgs`, along with the `ros1_bridge`.
+
+To create and build the workspace:
+
+1. Create the ROS 1 workspace directory using:
+   ```sh
+   $ mkdir -p ~/px4_ros_com_ros1/src
+   ```
+1. Clone the ROS 1 bridge packages `px4_ros_com` and `px4_msgs` to the `/src` directory (the `ros1` branch):
+   ```sh
+   $ git clone https://github.com/PX4/px4_ros_com.git ~/px4_ros_com_ros1/src/px4_ros_com -b ros1 # clones the 'ros1' branch
+   $ git clone https://github.com/PX4/px4_msgs.git ~/px4_ros_com_ros1/src/px4_msgs -b ros1
+   ```
+1. Use the `build_ros1_bridge.bash` script to build the ROS workspace (including `px4_ros_com`, `px4_msgs`, and `ros1_bridge`).
+   <!-- we didn't clone `ros1_bridge` ? -->
```

Comment:
How did the ros1_bridge get obtained - there was no instruction to clone it. Part of building with the script?

Reply:
@TSC21 This question is for you

Reply:
It's on line 40, it essentially says "you will need to create a separate workspace." but it doesn't really explain what this means.

It should probably be extended to say ", clone the repository and checkout the ros1 branch on a new directory"